metadata-models/src/main/pegasus/com/linkedin/mxe/SystemMetadata.pdl

```diff
@@ -38,4 +38,14 @@ record SystemMetadata {
   * Additional properties
   */
   properties: optional map[string, string]
+
+  /**
+  * The client id that produced the metadata.
+  */
+  clientId: optional string
```

Comment:
what is a client id? do we have that defined?

Reply:
We shouldn't need to extend the strongly typed metadata model for this use-case - feel free to use the properties key-value map to store these details.

Reply:
Will move to use properties